src/net/connection/connection_pool.rs

```diff
@@ -88,13 +96,16 @@ impl ConnectionPool {
 
     /// Check if there are any connections that have not been active for the given Duration.
     fn check_for_timeouts(
-        connections: &Connections,
+        connections: &mut Connections,
         sleepy_time: Duration,
         events_sender: &Sender<Event>,
     ) {
+        let mut timed_out_clients: Vec<SocketAddr> = Vec::with_capacity(connections.len());
```

Comment:
We should create this with an empty capacity, since we won't be removing connections that often.